plugins/performance-lab/includes/admin/rest-api.php

```diff
@@ -0,0 +1,170 @@
+					'type'              => 'string',
+					'description'       => __( 'Plugin slug of plugin/feature that needs to be activated.', 'performance-lab' ),
+					'required'          => true,
+					'validate_callback' => 'perflab_validate_slug_endpoint_arg',
+				),
+			),
+			'callback'            => 'perflab_handle_feature_activation',
+			'permission_callback' => static function () {
+				if ( current_user_can( 'install_plugins' ) ) {
+					return true;
+				}
+
+				return new WP_Error( 'cannot_install_plugin', __( 'Sorry, you are not allowed to install plugins on this site.', 'performance-lab' ) );
+			},
+		)
+	);
+
+	register_rest_route(
+		PERFLAB_REST_API_NAMESPACE,
+		PERFLAB_FEATURES_INFORMATION_ROUTE,
+		array(
+			'methods'             => 'GET',
+			'args'                => array(
+				'slug' => array(
+					'type'              => 'string',
+					'description'       => __( 'Plugin slug of plugin/feature whose information is needed.', 'performance-lab' ),
+					'required'          => true,
+					'validate_callback' => 'perflab_validate_slug_endpoint_arg',
+				),
+			),
+			'callback'            => 'perflab_handle_get_feature_information',
+			'permission_callback' => static function () {
+				if ( current_user_can( 'manage_options' ) ) {
+					return true;
+				}
+
+				return new WP_Error( 'cannot_access_plugin_settings_url', __( 'Sorry, you are not allowed to access plugin/feature information on this site.', 'performance-lab' ) );
+			},
+		)
+	);
+}
+add_action( 'rest_api_init', 'perflab_register_endpoint' );
+
+/**
+ * Validates whether the provided plugin slug is a valid Performance Lab plugin.
+ *
+ * Note that an enum is not being used because additional PHP files have to be required to access the necessary functions,
+ * and this would not be ideal to do at rest_api_init.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @param string $slug Plugin slug.
+ * @return bool Whether valid.
+ */
+function perflab_validate_slug_endpoint_arg( string $slug ): bool {
+	require_once ABSPATH . 'wp-admin/includes/plugin.php';
+	require_once PERFLAB_PLUGIN_DIR_PATH . 'includes/admin/load.php';
+	require_once PERFLAB_PLUGIN_DIR_PATH . 'includes/admin/plugins.php';
+	return in_array( $slug, perflab_get_standalone_plugins(), true );
+}
+
+/**
+ * Handles REST API request to activate plugin/feature.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @phpstan-param WP_REST_Request<array<string, mixed>> $request
+ *
+ * @param WP_REST_Request $request Request.
+ * @return WP_REST_Response|WP_Error Response.
+ */
+function perflab_handle_feature_activation( WP_REST_Request $request ) {
+	require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+	require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+	require_once ABSPATH . 'wp-admin/includes/class-wp-ajax-upgrader-skin.php';
+
+	// Install and activate the plugin/feature and its dependencies.
+	$result = perflab_install_and_activate_plugin( $request['slug'] );
```

Comment:
What about just short-circuiting with a `200` response? It would be a bad user experience if someone has two instances of the Performance screen open and they activate features in one tab, and then later try to activate features in the other tab. When they try to do so on the other tab, they'd find they can't activate the features even though they are already active.

In fact, `perflab_install_and_activate_plugin()` will already essentially skip doing anything if the plugin is already installed and active, so I don't think anything needs to be done here.

Reply:
Ok, that works too.